test/com/facebook/buck/rules/macros/ClasspathAbiMacroExpanderTest.java

```diff
@@ -0,0 +1,198 @@
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.rules.macros;
+
+import static com.facebook.buck.rules.TestCellBuilder.createCellRoots;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import com.facebook.buck.io.filesystem.ProjectFilesystem;
+import com.facebook.buck.jvm.java.JavaLibraryBuilder;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.BuildTargetFactory;
+import com.facebook.buck.model.macros.MacroException;
+import com.facebook.buck.rules.BuildRule;
+import com.facebook.buck.rules.BuildRuleResolver;
+import com.facebook.buck.rules.BuildableSupport;
+import com.facebook.buck.rules.CellPathResolver;
+import com.facebook.buck.rules.DefaultSourcePathResolver;
+import com.facebook.buck.rules.DefaultTargetNodeToBuildRuleTransformer;
+import com.facebook.buck.rules.FakeSourcePath;
+import com.facebook.buck.rules.SingleThreadedBuildRuleResolver;
+import com.facebook.buck.rules.SourcePathRuleFinder;
+import com.facebook.buck.rules.TargetGraph;
+import com.facebook.buck.rules.TargetNode;
+import com.facebook.buck.rules.args.Arg;
+import com.facebook.buck.shell.ExportFileBuilder;
+import com.facebook.buck.testutil.FakeProjectFilesystem;
+import com.facebook.buck.testutil.TargetGraphFactory;
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Optional;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ClasspathAbiMacroExpanderTest {
+  private ClasspathAbiMacroExpander expander;
+  private ProjectFilesystem filesystem;
+
+  @Before
+  public void createMacroExpander() {
+    this.expander = new ClasspathAbiMacroExpander();
+    this.filesystem = FakeProjectFilesystem.createJavaOnlyFilesystem();
+  }
+
+  @Test
+  public void shouldIncludeARuleIfNothingIsGiven() throws Exception {
+    final BuildRuleResolver ruleResolver =
+        new SingleThreadedBuildRuleResolver(
+            TargetGraph.EMPTY, new DefaultTargetNodeToBuildRuleTransformer());
+
+    BuildRule rule =
+        getLibraryBuilder("//cheese:cake")
+            .addSrc(Paths.get("Example.java")) // Force a jar to be created
+            .build(ruleResolver, filesystem);
+
+    assertExpandsTo(
+        rule,
+        ruleResolver,
+        String.format(
+            "%s" + File.separator + "%s",
+            filesystem.getRootPath(),
+            "buck-out/gen/cheese/cake#class-abi/cake-abi.jar"));
```

Comment:
These tests are failing on Windows. Our CI doesn't run these tests for some reason (we're investigating), but it looks like it's failing because of the path separators.

Reply:
using `new File.("file-path-string")toPath()` now, hopefully this would work fine on windows.